packages/flutter/lib/src/material/text_field.dart

```diff
@@ -985,7 +990,7 @@ class _TextFieldState extends State<TextField> with RestorationMixin implements
   final GlobalKey<EditableTextState> editableTextKey = GlobalKey<EditableTextState>();
 
   @override
-  bool get selectionEnabled => widget.selectionEnabled;
+  bool get selectionEnabled => widget.selectionEnabled && _isEnabled;
```

Comment:
Why was this change needed?

Reply:
This change was made because the text selection menu (copy, paste, etc...) opens up when right-clicking on the textfield when disabled (and ignorePointers set to false). So the selectionEnabled should take into account for _isEnabled to prevent selection in all cases when the widget is disabled.

Reply:
Ah makes sense, thanks for the explanation.